### PR TITLE
Homepage categorypage data

### PR DESCRIPTION
**Pull request by manojsarna**
- [x] Resources Added
- [x] Components Added - Header, Footer, Carousel, Category, Card
- [x] Routes Added 
- [x] Dark Theme Added
- [x] Mockman Integration
- [x] Homepage Added
- [x] Category Page Added

###Check these Files 

- [ ] [CategoryPage.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/homepage-categorypage-data?expand=1#diff-f7960e88b3ded03175d9efed0cb732f4b489f789a400ff80b75f4c7941d888a8)

- [ ] [Home.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/homepage-categorypage-data?expand=1#diff-172462f7215de52769ba39af6ddb83fcc09fcaefa36a75651b75c68a0e43a05a)

- [ ] [Videos.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/homepage-categorypage-data?expand=1#diff-1a6a8a136c50222a916915a6f25d5936d8d2caba5f646f3b48b0f7bcd9f710a9)

### Ignore Rest of the Files

image
image
image

### CODE REVIEW

File: src/pages/home/components/category/Category.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { useNavigate } from "react-router-dom";
+
+export function Category({ categoryItem }) {
+  const navigate = useNavigate();
+  return (
+    <div
+      className="sm-main-cat-child sm-box-shad-hov"
+      data-title={categoryItem.categoryName}
+      onClick={() => navigate(`/videos/${categoryItem.categoryName}`)}
```
Comment (rahulyadav139) [inline src/pages/home/components/category/Category.jsx:9]: use Link instead of navigate if you didn't plan to run any codes before.

**Review by rahulyadav139 — APPROVED** [review]

File: src/pages/video-page/VideoPage.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  const { videoId } = useParams();
+  const videoDetails = videos.find(
+    (video) => video.snippet.resourceId.videoId === videoId
+  );
+
+  const { pathname } = useLocation();
+
+  useEffect(() => {
+    window.scrollTo(0, 0);
+  }, [pathname]);
```
Comment (rahulyadav139) [inline src/pages/video-page/VideoPage.jsx:21]: This is a neat way to scroll to the top for different videos. Great work.

**Review by rahulyadav139 — APPROVED** [review]

File: src/pages/home/components/carousel/Carousel.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  const [imgSrc, setImgSrc] = useState(sliderImages[0]);
+  let i = 0;
+  useEffect(() => {
+    const interval = setInterval(() => {
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      if (i > sliderImages.length - 1) i = 0;
+      setImgSrc(() => sliderImages[i]);
+      i++;
+    }, 3000);
+    return () => clearInterval(interval);
```
Comment (dev-ashishsethi) [inline src/pages/home/components/carousel/Carousel.jsx:16]: clean logic

File: src/pages/video-page/VideoPage.jsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    <main className="sm-main-video-page">
+      <div className="sm-video-container">
+        <div className="sm-video-container-main">
+          <SmashPlayer videoId={videoId} />
+          <div className="sm-smash-player-cta-info">
+            <div className="sm-smash-player-info">
+              <span className="sm-smash-player-info-views">
+                {videoDetails.snippet.viewsCount} views
+              </span>
+              &nbsp;&nbsp;<span className="sm-smash-player-info-dot"></span>
```
Comment (dev-ashishsethi) [inline src/pages/video-page/VideoPage.jsx:35]: please don't use nbsp

**Review by dev-ashishsethi — APPROVED** [review]

File: src/pages/video-page/VideoPage.jsx (hunk 4 above)
Comment (manojsarna) [reply src/pages/video-page/VideoPage.jsx:35]: will update

**Pull request merged**